src/main/resources/com/google/api/codegen/ruby/message.snip

```diff
@@ -1,92 +1,92 @@
 @extends "ruby/common.snip"
 
-@snippet generateFilename(file)
-  doc/{@context.getBasename(file)}.rb
+@snippet generate(doc)
+  {@generateClass(doc)}
 @end
 
-@snippet generateClass(file)
-  @let body = documentChildren(file)
-    {@licenseSection()}
+@private generateClass(doc)
+  @let body = documentChildren(doc)
+    {@licenseSection(doc.fileHeader)}
 
-    {@module(context.getGrpcModules(file).iterator, body) }
+    {@module(util.getApiModules(doc.fileHeader.packageName).iterator, body)}
```

Comment:
There's some weird recursion in the snippet that I was having trouble figuring out how to move to the view model.